src/mahoji/lib/abstracted_commands/tinkeringWorkshopCommand.ts

```diff
@@ -0,0 +1,47 @@
+import { Time } from 'e';
+
+import { MaterialType, materialTypes } from '../../../lib/invention';
+import { transactMaterialsFromUser } from '../../../lib/invention/inventions';
+import { MaterialBank } from '../../../lib/invention/MaterialBank';
+import { TinkeringWorkshopOptions } from '../../../lib/types/minions';
+import { formatDuration, randomVariation } from '../../../lib/util';
+import addSubTaskToActivityTask from '../../../lib/util/addSubTaskToActivityTask';
+import { calcMaxTripLength } from '../../../lib/util/calcMaxTripLength';
+
+export async function tinkeringWorkshopCommand(user: MUser, material: MaterialType, channelID: string) {
+	if (!materialTypes.includes(material)) {
+		return "That's not a valid material.";
+	}
+	if (user.minionIsBusy) return 'Your minion is busy.';
+
+	const gameTime = Time.Minute * 12.5;
+	const quantity = Math.floor(calcMaxTripLength(user, 'TinkeringWorkshop') / gameTime);
+	const duration = randomVariation(quantity * gameTime, 5);
```

Comment:
I like to add a random duration to each quantity, and then sum them. This provides a more accurate + natural bell shaped curve of durations, instead of a random duration between 95 and 105% duration.

But it's not a big deal, especially for smaller quantities.